### ISSUE TITLE
Training aborted after few epochs accidentally

### ISSUE THREAD

**Issue by shaikhzhas** [facebookresearch/wav2letter#199]
I am training on Kazakh/Russian speech data

Here is log:
epoch:        1 | lr: 0.150000 | lrcriterion: 0.000000 | runtime: 00:16:34 | bch(ms): 39.79 | smp(ms): 6.77 | fwd(ms): 18.86 | crit-fwd(ms): 14.89 | bwd(ms): 8.77 | optim(ms): 3.88 | loss:   15.45428 | train-TER: 60.02 | data/dev-TER: 44.57 | avg-isz: 569 | avg-tsz: 091 | max-tsz: 410 | hrs:  158.12 | thrpt(sec/sec): 572.25
epoch:        2 | lr: 0.150000 | lrcriterion: 0.000000 | runtime: 00:15:44 | bch(ms): 37.79 | smp(ms): 6.71 | fwd(ms): 18.17 | crit-fwd(ms): 14.60 | bwd(ms): 8.38 | optim(ms): 3.33 | loss:   10.26184 | train-TER: 40.57 | data/dev-TER: 38.39 | avg-isz: 569 | avg-tsz: 091 | max-tsz: 410 | hrs:  158.12 | thrpt(sec/sec): 602.55
epoch:        3 | lr: 0.150000 | lrcriterion: 0.000000 | runtime: 00:15:54 | bch(ms): 38.18 | smp(ms): 6.85 | fwd(ms): 18.39 | crit-fwd(ms): 14.79 | bwd(ms): 8.32 | optim(ms): 3.37 | loss:    8.89558 | train-TER: 35.28 | data/dev-TER: 34.36 | avg-isz: 569 | avg-tsz: 091 | max-tsz: 410 | hrs:  158.12 | thrpt(sec/sec): 596.41
epoch:        4 | lr: 0.150000 | lrcriterion: 0.000000 | runtime: 00:15:41 | bch(ms): 37.66 | smp(ms): 6.67 | fwd(ms): 18.24 | crit-fwd(ms): 14.73 | bwd(ms): 8.24 | optim(ms): 3.33 | loss:    8.14123 | train-TER: 32.34 | data/dev-TER: 34.63 | avg-isz: 569 | avg-tsz: 091 | max-tsz: 410 | hrs:  158.12 | thrpt(sec/sec): 604.64
epoch:        5 | lr: 0.150000 | lrcriterion: 0.000000 | runtime: 00:15:41 | bch(ms): 37.64 | smp(ms): 6.68 | fwd(ms): 18.22 | crit-fwd(ms): 14.71 | bwd(ms): 8.21 | optim(ms): 3.35 | loss:    7.62757 | train-TER: 30.40 | data/dev-TER: 31.39 | avg-isz: 569 | avg-tsz: 091 | max-tsz: 410 | hrs:  158.12 | thrpt(sec/sec): 604.85
epoch:        6 | lr: 0.150000 | lrcriterion: 0.000000 | runtime: 00:15:45 | bch(ms): 37.82 | smp(ms): 6.86 | fwd(ms): 18.26 | crit-fwd(ms): 14.69 | bwd(ms): 8.18 | optim(ms): 3.32 | loss:    7.24471 | train-TER: 28.96 | data/dev-TER: 31.71 | avg-isz: 569 | avg-tsz: 091 | max-tsz: 410 | hrs:  158.12 | thrpt(sec/sec): 602.07
epoch:        7 | lr: 0.150000 | lrcriterion: 0.000000 | runtime: 00:15:41 | bch(ms): 37.67 | smp(ms): 6.77 | fwd(ms): 18.26 | crit-fwd(ms): 14.66 | bwd(ms): 8.21 | optim(ms): 3.27 | loss:    6.93145 | train-TER: 27.82 | data/dev-TER: 31.44 | avg-isz: 569 | avg-tsz: 091 | max-tsz: 410 | hrs:  158.12 | thrpt(sec/sec): 604.48
epoch:        8 | lr: 0.150000 | lrcriterion: 0.000000 | runtime: 00:15:40 | bch(ms): 37.64 | smp(ms): 6.66 | fwd(ms): 18.32 | crit-fwd(ms): 14.72 | bwd(ms): 8.20 | optim(ms): 3.30 | loss:    6.66556 | train-TER: 26.87 | data/dev-TER: 29.78 | avg-isz: 569 | avg-tsz: 091 | max-tsz: 410 | hrs:  158.12 | thrpt(sec/sec): 604.94
epoch:        9 | lr: 0.150000 | lrcriterion: 0.000000 | runtime: 00:15:35 | bch(ms): 37.43 | smp(ms): 6.61 | fwd(ms): 18.22 | crit-fwd(ms): 14.64 | bwd(ms): 8.16 | optim(ms): 3.27 | loss:    6.42914 | train-TER: 26.01 | data/dev-TER: 30.50 | avg-isz: 569 | avg-tsz: 091 | max-tsz: 410 | hrs:  158.12 | thrpt(sec/sec): 608.30

After 9th epoch it stopped abruptly:
*** Aborted at 1549607938 (unix time) try "date -d @1549607938" if you are using GNU date ***
PC: @     0x7fbe467dc227 (unknown)
*** SIGTERM (@0x7fffb2e05f68) received by PID 30106 (TID 0x7fbeb14d8dc0) from PID 18446744072415633256; stack trace: ***
    @     0x7fbe4672b060 (unknown)
    @     0x7fbe467dc227 (unknown)
    @     0x7fbe4a133458 std::__atomic_futex_unsigned_base::_M_futex_wait_until()
    @     0x56199ec2040a w2l::W2lDataset::getFeatureDataAndPrefetch()
    @     0x56199ec20b5b w2l::W2lDataset::get()
    @     0x56199eb102bc _ZZ4mainENKUlSt10shared_ptrIN2fl6ModuleEES_IN3w2l17SequenceCriterionEES_INS3_10W2lDatasetEES_INS0_19FirstOrderOptimizerEES9_ddbiE4_clES2_S5_S7_S9_S9_ddbi
    @     0x56199eacd7b2 main
    @     0x7fbe467182e1 __libc_start_main
    @     0x56199eb0c31a _start
    @                0x0 (unknown)

I don't have a clue what can be cause of this issue?

Best

**Comment by jacobkahn** [comment]
It looks like your loss is decreasing, so things are generally working. This looks like an issue with reading in/prefetching the dataset, and could be related to memory. A few questions:
- Can you reproduce the issue? (does it always break here?)
- What OS and CUDA version/GPU type do you have/how much GPU memory?
- What are the relative dimensions of your dataset (how many audio files, and how long is each file?)

**Comment by RVlasov** [comment]
Have you successfully pretrained your model? I'm a trying to start wav2letter with russian. Maybe you can help how to collect russian dataset the best way? Ore you have used prepared dataset?

**Comment by shaikhzhas** [comment]
> It looks like your loss is decreasing, so things are generally working. This looks like an issue with reading in/prefetching the dataset, and could be related to memory. A few questions:
> 
> * Can you reproduce the issue? (does it always break here?)
> * What OS and CUDA version/GPU type do you have/how much GPU memory?
> * What are the relative dimensions of your dataset (how many audio files, and how long is each file?)

Hi

Thanks for responding,
I have installed wav2letter on another CUDA, GPU type, but still is has the same problem.

Also, I have decreased my data size, but still the same problem.
Seems like it doesn't depend on the data size.

What I have noticed is that the number of epochs when it crashes is not different each time I run the  model. I don't have any clue to explain this problem.

Please, anyone help me.
